docs: add new count and range subfields to manifest docs

SoManyHs opened the pull request:
Closes https://github.com/aws/copilot-cli/issues/2205

NOTE: This change also includes a refactor that makes use of the `mkdocs-macros` pluging. This allows embedding markdown files into other ones, which allows us to de-dupe the docs for the two types of service manifests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

SoManyHs committed:
docs: add new count and range subfields to manifest docs

SoManyHs committed:
chore: remove unused Makefile reference

SoManyHs commented:
What it looks like:

<img width="715" alt="Screen Shot 2021-04-22 at 15 36 15" src="https://user-images.githubusercontent.com/29964746/115793407-ebac2180-a380-11eb-8df9-4ce7f7a17653.png">

SoManyHs committed:
docs: add example of spot count

SoManyHs commented:
What the example manifest looks like (added to env overrides):

<img width="800" alt="Screen Shot 2021-04-22 at 15 42 41" src="https://user-images.githubusercontent.com/29964746/115793680-6412e280-a381-11eb-879d-1ef78b83e77a.png">

Lou1415926 reviewed (commented):
Super clear and easy to understand!! Can we also add the same doc for `manifest/backend-servuce` since it also applies there 🦃 🙇🏼‍♀️ ?

SoManyHs committed:
chore: strip whitespace

huanjani commented on site/content/docs/manifest/lb-web-service.md:
````diff
@@ -199,8 +208,34 @@ count:
+    spot_from: 3
+```
+
+This will set your range as 1-10 as before, but will place two copies of your service on dedicated Fargate capacity. Once your service scales to 3 copies, they will be placed on spot until the maximum is reached.
````
```suggestion
This will set your range as 1-10 as above, but will place the first two copies of your service on dedicated Fargate capacity. If your service scales to 3 or higher, the third and any additional copies will be placed on Spot until the maximum is reached.
```

huanjani commented on site/content/docs/manifest/lb-web-service.md:
````diff
@@ -199,8 +208,34 @@ count:
+```
+This will set up an Application Autoscaling Target with the `MinCapacity` of `n` and `MaxCapacity` of `m`.
+
+Alternatively, if you wish to scale your service onto Fargate Spot instances, specify `min` and `max` under `range` and then specify `spot_from` with the desired count you wish to start placing your services onto spot capacity. For example:
````
```suggestion
Alternatively, if you wish to scale your service onto Fargate Spot instances, specify `min` and `max` under `range` and then specify `spot_from` with the desired count you wish to start placing your services onto Spot capacity. For example:
```

huanjani commented:
Looks great, super clear! Just some tiny nits.

SoManyHs committed:
docs: add advanced count docs to backend svc manifest

SoManyHs committed:
Update site/content/docs/manifest/lb-web-service.md

Co-authored-by: Janice Huang <60631893+huanjani@users.noreply.github.com>

SoManyHs committed:
Update site/content/docs/manifest/lb-web-service.md

Co-authored-by: Janice Huang <60631893+huanjani@users.noreply.github.com>

SoManyHs committed:
chore: update wording

SoManyHs committed:
chore: change double brackets to carrots in schema examples

Double brackets {{ }} have a special meaning in jinja2 and this will
result in an error: "jinja2.exceptions.TemplateSyntaxError: expected
token 'end of print statement' ..." if used with macros.

SoManyHs committed:
refactor: DRY up manifest documentation

Use mkdir-macros in order to include markdown "partials" in other md
files.

See: https://mkdocs-macros-plugin.readthedocs.io/en/latest/advanced/

SoManyHs committed:
Merge branch 'mainline' into spot-docs

huanjani commented on site/content/docs/include/common-svc-fields.md:
````diff
@@ -0,0 +1,180 @@
+  response_time: 2s
+```
+
+<span class="parent-field">count.</span><a id="count-range" href="#count-range" class="field">`range`</a> <span class="type">String or Map</span>
````
Should this be `Integer or Map`? 
I believe ranges of ints like `1-5` are considered ints?

SoManyHs replied:
Sure, but the range as a whole is considered a string: https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/svc.go#L75

huanjani replied:
Ah, gotcha. I had checked with `reflect.TypeOf` but didn't see that we were handling these as strings. Thank you!

SoManyHs committed:
Merge branch 'mainline' into spot-docs

huanjani reviewed (approved):
This looks great! Thanks for DRYing up the docs with partials!
